### PR TITLE
feat(python): Display struct field's name and dtype

### PR DESCRIPTION
**Pull request by phillyphil91**
Related #3925
I know there is a lot missing here, but I wanted to start with the rust side first. As there is already a good idea how modify the schema method in python: #3953

This displays the structs field's name and dtype in a simple way, but formatting will fail for structs with further structs as fields. Display follows pyspark style from example in #3925 

Not sure if this is the best way going forward, as the result in python would look as follows (also check python test checks):
image

Instead of:
image

Hence all doctest examples would have to be adjusted. I can do that, but wanted to make sure this approach goes into the right direction.

### CODE REVIEW

**Comment by phillyphil91** [comment]
Hey @ritchie46 ,
any comment or suggestions on this? I think this might be mostly a stylistic change. But could also have some drastic changes when you print the schema. I would be happy to change the output to a desired format

**Comment by ghuls** [comment]
The full struct printing could be toggled by a pl.Config option similar to: `pl.Config.set_tbl_hide_column_data_types()` and other table related settings.

**Comment by phillyphil91** [comment]
thanks for the reply @ghuls. I will take a look at how to add this to `pl.Config `and re-open the PR when i have changed rebased on the current state of polars as well

**Comment by phillyphil91** [comment]
Hey @ghuls 
I have updated the code to allow a config flag to be set on the python side for the extensive displaying of the struct. Is that what you were thinking?
I have tested it and it works with the config flag. Without it, it displays the dtype the old way. Should there be any tests specifically for that? I have not seen any other tests for displaying/ printing the other dtypes, if i have seen correctly.  

A couple of notes:
With the new way of displaying it, it no longer gives the correct datatype of polars.datatypes.Utf8 for example but rather:

image

Also for nested structs, i.e. structs that contain structs this display breaks:
image

I'm not really sure how to fix this to be honest, as I don't know how to keep track of the level of nesting to increase the indentation or something in the `fmt` function. 

Let me know if you have any thoughts on this or can suggest something.

**Comment by phillyphil91** [comment]
Hey @ghuls any update or feedback on this? I'm not sure if the new formatting makes sense, but if we can agree on something than I could move forward with this.

File: polars/polars-core/src/datatypes/dtype.rs [hunk 1]
```diff
@@ -235,6 +236,10 @@ impl PartialEq<ArrowDataType> for DataType {
     }
 }
 
+fn env_is_true(varname: &str) -> bool {
```
Comment (ghuls) [inline polars/polars-core/src/datatypes/dtype.rs:239]: Can't it be reused from fmt.rs?

**Comment by ghuls** [comment]
The struct fields shouldn't be displayed in the schema, only in the printed dataframe. The schema is a python dictionary, the new formatting breaks this (and would also break code that manipulates the schema).

**Comment by phillyphil91** [comment]
is this still needed @ghuls ? 
I saw that df.schema basically displays the struct's fields now.

**Pull request closed without merging**